tests/templates/test_state_preparations/test_state_prep_mps.py

```diff
@@ -21,232 +21,1581 @@
+            ),
+            (
+                [
+                    np.array([[0.0, 0.107], [0.994, 0.0]]),
+                    np.array(
+                        [
+                            [[-1.0, 0.0], [0.0, 0.0]],
+                            [[0.0, 0.0], [0.0, 1.0]],
+                            [[0.0, -1.0], [0.0, 0.0]],
+                            [[0.0, 0.0], [1.0, 0.0]],
+                        ]
+                    ),
+                    np.array(
+                        [
+                            [[-1.0, 0.0], [0.0, 0.0]],
+                            [[0.0, 0.0], [0.0, 1.0]],
+                            [[0.0, -1.0], [0.0, 0.0]],
+                            [[0.0, 0.0], [1.0, 0.0]],
+                        ]
+                    ),
+                    np.array([[-1.0, -0.0], [-0.0, -1.0]]),
+                ],
+                "Dimension mismatch:",
+            ),
+            (
+                [
+                    np.array([[0.0, 0.107], [0.994, 0.0]]),
+                    np.array(
+                        [
+                            [[0.0, 0.0, 0.0, -0.0], [1.0, 0.0, 0.0, -0.0]],
+                            [[0.0, 1.0, 0.0, -0.0], [0.0, 0.0, 0.0, -0.0]],
+                        ]
+                    ),
+                    np.array(
+                        [
+                            [[-1.0, 0.0], [0.0, 0.0]],
+                            [[0.0, 0.0], [0.0, 1.0]],
+                            [[0.0, -1.0], [0.0, 0.0]],
+                            [[0.0, 0.0], [1.0, 0.0]],
+                        ]
+                    ),
+                    np.array([-1.0, -0.0, -0.0, -1.0]),
+                ],
+                "The last tensor must have exactly 2 dimensions.",
+            ),
+            (
+                [
+                    np.array([[0.0, 0.107], [0.994, 0.0]]),
+                    np.array(
+                        [
+                            [[0.0, 0.0, 0.0, -0.0], [1.0, 0.0, 0.0, -0.0]],
+                            [[0.0, 1.0, 0.0, -0.0], [0.0, 0.0, 0.0, -0.0]],
+                        ]
+                    ),
+                    np.array(
+                        [
+                            [[-1.0, 0.0], [0.0, 0.0]],
+                            [[0.0, 0.0], [0.0, 1.0]],
+                            [[0.0, -1.0], [0.0, 0.0]],
+                            [[0.0, 0.0], [1.0, 0.0]],
+                        ]
+                    ),
+                    np.array([[-1.0, -0.0, 0.0], [-0.0, -1.0, 0.0]]),
+                ],
+                "The second dimension of the last tensor must be exactly 2.",
+            ),
+        ],
+    )
+    def test_MPSPrep_error(self, mps, msg_match):
+        """Test that proper errors are raised for MPSPrep"""
+        with pytest.raises(AssertionError, match=msg_match):
+            qml.MPSPrep(mps, wires=[0, 1, 2])
+
+    @pytest.mark.jax
+    def test_jax_mps(self):
```

Comment:
Is the only difference between this test and the later JAX one the use of JIT?
Could we combine these, or just remove the non-Jitted one in favour of the other?